apps/discovery_streams/lib/discovery-streams/event_handler.ex

```diff
@@ -31,6 +31,25 @@ defmodule DiscoveryStreams.EventHandler do
     :ok
   end
 
+  def handle_event(%Brook.Event{
+        type: dataset_delete(),
+        data: %Dataset{id: id, technical: %{systemName: system_name}}
+      }) do
+    case delete_dataset(id) do
+      :ok ->
+        Logger.debug("#{__MODULE__}: Deleted dataset for dataset id: #{id} and system name: #{system_name}")
+
+      {:error, error} ->
+        Logger.error(
+          "#{__MODULE__}: Failed to delete dataset for dataset id: #{id} and system name: #{system_name}, Reason: #{
+            inspect(error)
+          }"
+        )
+    end
```

Comment:
Consider moving this whole block into its own block and calling it `delete_dataset_topic`?